### DISCUSSION TITLE
Creating a delay task that takes in liquid input

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#7136] [Q&A] [answered]
Hi guys, I'm trying to create a "delay" event which accepts liquid input.. but I cant really figure out how to do it.

I'm currently doing an event activity which does a Task.Delay and a background Task which calls the event. but this doesn't work as the Delay Event is reset when the delay event is triggered by my background service.

Here's my code at the moment:

```
        public override async Task<bool> CanExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
        {
            return await IsExpired(workflowContext);
        }

        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
        {
            return Outcomes(T["Done"]);
        }

        public override ActivityExecutionResult Resume(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
        {
            return Outcomes("Done");
        }

        private async Task<bool> IsExpired(WorkflowExecutionContext workflowContext)
        {
            var timeStart = DateTimeOffset.UtcNow;
            var duration = await _expressionEvaluator.EvaluateAsync(DelayTimeInMilliseconds, workflowContext, null);

            await Task.Delay((int)duration);
            var timeEnd = DateTimeOffset.UtcNow;
            workflowContext.LastResult = $"Started: {timeStart}; Ended: {timeEnd}; Duration {timeEnd.Subtract(timeStart)}";
            return true;
        }
```

I'm thinking I need to somehow parse the milliseconds passed in to a cron expression and follow the same pattern as the timer event.

Thank you for the help!

**Comment by Piedone** [comment]
I'm not sure I understand why you'd need this instead of the Timer event (though that operates in the order of magnitude of seconds and above).

**Reply by weirdyang** [reply]
Because I would like to do a post with a date string and be able to schedule a task. The current timer event does not accept a liquid input.
For example a http post with body:

```
{
    "TargetTime": "02-10-2020 10:00:55";
}
```
Anyway i've solved this issue!

**Reply by Piedone** [reply]
Ah, OK then, sorry I missed that. Good that you solved it!

Otherwise, I'd recommend copying `TimerEvent` and taking `NotifyTask` as an example of how to enable Liquid in it. Would you also open an issue about this? Maybe it'd be useful for others too and we can implement it eventually.

**Reply by weirdyang** [reply]
Sure. I've uploaded my solution [here](https://github.com/weirdyang/OrchardCore.Olaf), it definitely can be improved.

**Comment by weirdyang** [comment, marked as answer]
https://github.com/weirdyang/OrchardCore.Olaf

I solved It by implementing a new module
